lib/app/modules/bulk_data/bulk_data_patient_export_sequence.rb

```diff
@@ -59,13 +72,55 @@ def export_kick_off(klass)
           versions :stu3
         end
 
-        reply = export_kick_off('Patient')
+        assert_export_kick_off('Patient')
+      end
 
-        assert_response_accepted(reply)
+      test 'Server shall return "202 Accepted" or "200 OK"' do
+        metadata do
+          id '03'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-status-request'
+          desc %(
+          )
+          versions :stu3
+        end
 
-        @content_location = reply.response[:headers]['content-location']
+        assert_export_status
+      end
+
+      private
+
+      def export_kick_off(klass)
+        headers = { accept: 'application/fhir+json', prefer: 'respond-async' }
+
+        url = ''
+        url += "/#{klass}" if klass.present?
+        url += '/$export'
+
+        @client.get(url, @client.fhir_headers(headers))
+      end
+
+      def export_status_check(url, headers = { accept: 'application/json' })
+        wait_time = 1
+        reply = nil
+
+        loop do
+          reply = @client.get(url, @client.fhir_headers(headers))
+
+          break if reply.code != 202
+
+          retry_after = reply.response[:headers]['retry-after']
+          wait_time = (retry_after.presence || wait_time * 2).to_i
```

Comment:
What if they have no retry-after header?  This will cause `retry_after` to be nil, and `retry_after.presence` to raise an unhelpful "no presence method on nil on line xyz" error (vs. a nicely worded failure).  Similarly what if presence isn't a number.  Or if presense in in milliseconds but we expect seconds... will a huge number here cause this test to hang indefinitely?

Reply:
retry_after.presence return nil even if the 'retry-after' header does NOT exist.
Add a logic check if (current time used + wait time > timeout)